explore_utils.py

```diff
@@ -102,13 +102,34 @@ def find_channel_grouping_revenue(dataset):
 
     return counts, means
 
+
+def find_transaction_by_region(data):
+    """ Find the average transaction revenue by region
+
+    args:
+        dataset (Dataset): the google analytics Dataset
+
+    returns:
+        Dataframe of total transaction revenue by region in ascending order.
+
+    """
+    train_df = data.train
+    data = train_df.copy(deep=False)
+    data['rev'] = data['totals.transactionRevenue'].fillna(0).astype(float)
+    avg = data.groupby('geoNetwork.region')['rev'].mean()
+    avg = pd.DataFrame(avg)
+    new_df = avg[avg['rev']>0]
+    new_df.columns = ['Transaction']
+    return new_df.sort_values(by=['Transaction'])
+  
 def find_percentage_single_transaction(dataset):
 
-    train_df2 = dataset.train
+    train_df2 = dataset.train.copy()
     train_df2.set_index('fullVisitorId', inplace=True)
     continents = train_df2['geoNetwork.continent'].unique()
+    result = ""
     for con in continents:
         number_of_people = train_df2[train_df2['geoNetwork.continent'] == con]['visitNumber'].count()
-        train_df3 = train_df2[train_df2['geoNetwork.continent'] == con]['visitNumber']
-        number_of_single_visits = train_df3[train_df3 == 1].count()
-        print('Percentage of people visited for once in '+ con + ': '+(str)((number_of_single_visits/number_of_people)*100))
+        number_of_single_visits = train_df2[(train_df2['geoNetwork.continent'] == con) & (train_df2['visitNumber']== 1)]['visitNumber'].count()
+        result = result + 'Percentage of people visited for once in '+ con + ': '+(str)("%.2f" %((number_of_single_visits/number_of_people)*100)) +'\n'
```

Comment:
Rather than returning a string, you should return a value and take care of printing in `explore.py`.